### ISSUE TITLE
[Patreon Goal] Humble Bundle integration 

### ISSUE THREAD

**Issue by strycore** [lutris/lutris#334]
I know we planned to use code based on humblebundle-python but I'm not sure it's really maintained anymore. Instead, this looks really interesting: https://github.com/MestreLion/humblebundle

**Comment by RobLoach** [comment]
Looks like a great project. Both API, and command line usage too.

**Comment by RobLoach** [comment]
https://github.com/MestreLion/humblebundle

**Comment by sparr** [comment]
Would this functionality be expected to work similar to the GoG integration, allowing the user to log into their humble account to download?

**Comment by strycore** [comment]
Yes. Why are you asking?

**Comment by FruityWelsh** [comment]
Gamehub supports humble trove/bundle pretty seamlessly : https://github.com/tkashkin/GameHub

Maybe there is something we could gleam there?

**Comment by frio** [comment]
Actually -- yeah, there is some interesting stuff in there.  So far all the Humble libraries I've personally had a hack on just scrape the website and parse the HTML for data, but that integration uses an API.

The relevant endpoints are:

 - https://www.humblebundle.com/api/v1/user/order?ajax=true -- all the orders the authenticated user has made
 - https://www.humblebundle.com/api/v1/order/${key}?ajax=true -- information about the requested order, including download links when applicable

Super cool.  That should make building an integration reasonably straightforward, and likely without the need for an external library.  If I get some time this weekend (big if), I might hack something out and submit it.

**Comment by frio** [comment]
Just for reference, other API consumers: https://github.com/UncleGoogle/galaxy-integration-humblebundle/blob/master/src/plugin.py, https://github.com/MayeulC/hb-downloader.  I've bodged something (bad) together that at least fetches all the games a user owns (it doesn't merge with Lutris' internal IDs yet) -- but to authenticate, a user has to login in their browser and dump a cookie.

I'm relatively happy doing that personally, but I don't know what level of UX polish Lutris wants to present to people; is that an acceptable experience? For instance -- alternatively, should we embed a WebKitGTK and dump the cookie from that?  Is that acceptable as a long-term goal, with manual cookie-dumping a short term connivance (maybe stashed behind a feature flag)?
